src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -382,31 +389,48 @@ private function process_directives_args( string $html, array &$context_stack, a
 				}
 			}
 		}
-
 		/*
 		 * It returns null if the HTML is unbalanced because unbalanced HTML is
 		 * not safe to process. In that case, the Interactivity API runtime will
-		 * update the HTML on the client side during the hydration.
+		 * update the HTML on the client side during the hydration. It will also
+		 * display a notice to the developer to inform them about the issue.
 		 */
-		return $unbalanced || 0 < count( $tag_stack ) ? null : $p->get_updated_html();
+		if ( $unbalanced || 0 < count( $tag_stack ) ) {
+			$tag_errored = 0 < count( $tag_stack ) ? end( $tag_stack )[0] : $tag_name;
+			/* translators: %s: Tag that caused the error, could by any HTML tag. */
+			$message = sprintf( __( 'Interactivity directives failed to process in "%1$s" due to a missing "%2$s" end tag.' ), end( $namespace_stack ), $tag_errored );
+			_doing_it_wrong( __METHOD__, $message, '6.6.0' );
+			return null;
+		}
+
+		return $p->get_updated_html();
 	}
 
 	/**
 	 * Evaluates the reference path passed to a directive based on the current
 	 * store namespace, state and context.
 	 *
 	 * @since 6.5.0
+	 * @since 6.6.0 The function now adds a warning when the namespace is null, falsy, or the directive value is empty.
 	 *
 	 * @param string|true $directive_value   The directive attribute value string or `true` when it's a boolean attribute.
 	 * @param string      $default_namespace The default namespace to use if none is explicitly defined in the directive
 	 *                                       value.
 	 * @param array|false $context           The current context for evaluating the directive or false if there is no
 	 *                                       context.
-	 * @return mixed|null The result of the evaluation. Null if the reference path doesn't exist.
+	 * @return mixed|null The result of the evaluation. Null if the reference path doesn't exist or the namespace is falsy.
 	 */
 	private function evaluate( $directive_value, string $default_namespace, $context = false ) {
 		list( $ns, $path ) = $this->extract_directive_value( $directive_value, $default_namespace );
-		if ( empty( $path ) ) {
+		if ( 'null' === $default_namespace ) {
```

Comment:
Is this supposed to be the value `null` or the string `"null"`? It's the string now which looks pretty strange.

Reply:
It is strange, but is the string. Is read as a valid string namespace for SSR but then crashes in the runtime (need to 100% confirm this)

Reply:
The JS runtime calls JSON.parse on directives… maybe it should only do that in directives that look like objects.

I assume many stringified JS could make that crash now.

Reply:
I think any strings should be valid namespaces. null shouldn't be special.

Seems like a bug.

Reply:
Then I'll try to fix it! 🚀

Reply:
Using numbers only triggers the same error 😅